### ISSUE TITLE
Multiple TSLint rules fail with non-null assertions

### ISSUE THREAD

**Issue by Kokogino** [palantir/tslint#4018]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.7.2
- __Running TSLint via__: Angular CLI -> Node.js API

#### TypeScript code being linted

```ts
dupLoginid!: Observable<boolean>;

invalidFirstName!: Observable<boolean>;

invalidLastName!: Observable<boolean>;
```

with `tslint.json` configuration:

```json
"member-ordering": [
  true,
  {
    "order": [
      "static-field",
      "instance-field",
      "static-method",
      "instance-method"
    ]
  }
],
"semicolon": [
  true,
  "always"
],
"whitespace": [
  true,
  "check-branch",
  "check-decl",
  "check-operator",
  "check-separator",
  "check-type"
]
```

#### Actual behavior
TSLint complains about:
* "missing whitespace (whitespace)" before the `!`s
* "Missing semicolon (semicolon)" after the `!`s
* "Declaration of instance field not allowed after declaration of instance method. Instead, this should come at the beginning of the class/interface. (member-ordering)" on all but the first attribute.

#### Expected behavior
No TSLint complaints.

**Comment by JoshuaKGoldberg** [comment]
@Kokogino I'm not able to reproduce these errors locally with TSLint 5.11. `!` were pretty new when you posted this issue, perhaps your tooling was using an older version of TypeScript at the time?
